tests/functional/conftest.py

```diff
@@ -106,7 +114,16 @@ def _check(container):
 
 
 @pytest.fixture
-def wait_for_sidekiq(gl):
+def wait_for_sidekiq(gl: gitlab.Gitlab):
+    """
+    Return a helper function to wait until there are no busy sidekiq processes.
+
+    Use this with asserts for slow tasks (group/project/user creation/deletion).
+    """
+    return _wait_for_sidekiq(gl=gl)
+
+
+def _wait_for_sidekiq(gl: gitlab.Gitlab):
```

Comment:
> I'm trying to see why we'd need the 3 levels of nesting here. I think it's because a fixture function is being used in a helper function. Maybe we can avoid this by:

After doing some different work i realize I could have just made it 2 levels.

> a) converting the other helper into a fixture if it makes sense, and pass this one explicitly
> b) rely on helpers only and no fixtures for this 
> c) remove `reset_gitlab` helper (seems to be the problematic one) and just do the cleanup logic inside the `gl` fixture which makes the most sense in a way: the fixture would prepare and then return the gitlab instance. I think `reset_gitlab` was copied from a script from before pytest was used anyway.
> 
> c) seems cleanest to me. Then we'd also avoid having to call things with `_wait_for_sidekiq(gl=gl)(timeout=60)` etc.

Yeah we could do that. I've actually been working on a different PR to lessen use of `wait_for_sidekiq` as it really doesn't work that well, unfortunately. I mean it waits but even though nothing is busy we may still have things that need to finish happening 😟

Reply:
Ahh this flakiness is annoying, sorry :grimacing: ok if we can continue with putting `reset_gitlab` into `gl` that might be a start for now and look at it more in the other PR.

Reply:
Ah. I think I know why I have 3-levels here. Because I need the `gl` value. If that makes sense.

Reply:
Okay. I moved the functionality of `reset_gitlab` into the `gl` fixture and deleted `reset_gitlab`.

Reply:
I figured out a way to make the `wait_for_sidekiq` a bit cleaner...

Reply:
This is resolved as after rebasing I didn't need it to change `wait_for_sidekiq` any more.